hooks/checkout

```diff
@@ -99,19 +104,23 @@ checkout() {
 
   if [[ -z "${BUILDKITE_COMMIT:-}" || "${BUILDKITE_COMMIT}" == "HEAD" ]]; then
     exit_code=0
-    timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v --no-tags origin "${BUILDKITE_BRANCH}" || exit_code=$?
+    echo "---------- "${GIT_REMOTE_TIMEOUT}" GIT_TRACE=1 GIT_CURL_VERBOSE=1 git fetch -v --no-tags origin ${BUILDKITE_BRANCH} || exit_code=$? ---------- " >> "${log}"
+    (GIT_TRACE=1 GIT_CURL_VERBOSE=1 timeout "${GIT_REMOTE_TIMEOUT}"git fetch -v --no-tags origin "${BUILDKITE_BRANCH}" || exit_code=$?)  &>> "${log}"
```

Comment:
will this hide any output from the output window in buildkite?, I think we should send stdout to the console and stderr to the file.

Reply:
actually seems like we can set the file as the trace env var, probably this will be easier, something like
```
GIT_TRACE=${log}
```
not sure if the same works for `GIT_CURL_VERBOSE`
https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables

Reply:
this will only pipe STDERR to a file, so STDOUT will still go to the console.

I like setting the env var to point to a file though, so I'll modify the PR to use this

Reply:
It seems there's still quite a lengthy trace posted to STDERR that's not uploaded to the file using this: https://buildkite.com/canva-org/canva-web-build/builds/386828#fca1fe0c-1044-4733-873d-04b67bc3b700/82-112

I'm not sure if what's in the[ file](https://s3.amazonaws.com/buildkiteartifacts.com/358049d4-baf1-4853-8e46-eb5761e07794/f1ad1196-926b-43c9-a9e0-c80420302e60/ee7cfdb3-6c63-4389-9f06-df0693aca0ae/fca1fe0c-1044-4733-873d-04b67bc3b700/tmp/githubtrace.fZDVMkvitT.log?response-content-type=text%2Fplain&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7L5WCDRCNJ%2F20210208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210208T045944Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEHQaCXVzLWVhc3QtMSJHMEUCIQDawjNVsc5A0IB1zAI4RJ554Xv0C%2Bx%2BJLdlj6cpzJTuKAIgcEi5I9OCusVa5xsCyMIUvmizKTm5FUliDS3wH%2BBvwO4qtAMIbRAAGgwwMzIzNzk3MDUzMDMiDFXHDFvf2BELVyaFvCqRA55lCMBcEpSgd1RcG5zazdhFJWUAQIBNO%2FM3LpAYJldgJNhlUvi%2BidR791ey2hhwo7pKCs16iqOdyIwToFW69kDvlj5md%2FaAr358q8Ka3cI79vbCdZCWfCCQuNkHUjXlq2dg22VN7pOEnFzr7T%2F0HDpQolTm4r6ULZ0vXIXY2O9RGbOVUUK53MiHoEqd%2BAmlOupM%2FfpF7HMxLdJ5%2BfM384XIFOJBeckrSDiuuUcp6%2BQ27Tq3E5cLe4VNaU3u6JeqIJkH%2F%2FCkyYJrNQr9yrYkxAbmN%2FwSrvFi3rY6klwjvFSxTeA9rrIDqPX33Ihowbudrr7LpEtdfN5Jq%2FgJr8YHIZ7zQPMFN5y5Kf%2BMIQxIUymTQVe9YtZy6fSSdItWVUWeQGbexi2Ko2gT2eGBefm1cgrvFeoiNy0QhpgcJxPQDnhQgY2aWNuikR38puYloFNQI%2BdFUfGYOHJ8T1C%2Fc8Zd2OwIuVxl7bOyuE0EASjhNDeDhItr%2B4XeRhFn5zRuvMNZu%2BO2a44hvakHKp9pLla9MFPeMKnigoEGOusB7XA%2BPp1Rns6FYkTU451SkFo4niJyHfFDoDY1Pe0AC9Muv1zLzNmLjjkULmIaEiECPmJwy4Sdkj3%2BjP%2FelLM1JK7rCDpNF1Nq%2FCLUHQxFDIcqJxVzCwBkNahO6WOYpyF50jXUVj3FzZ73grU2xn79rAnXd5SQlkGpiLqJ1qtX868NC%2FQIcy8wVQMkHzEBJzl4xyRaA11Lw%2BQ9CzQTKmpeBiYU5Sj40XDjVhxGLhc3B7Dw48z9YYxL%2BZxk%2FpHTVqRQbIE5NxWnc1Ub7Y%2Bx8XgCDO3%2B%2F6j4uV9qFf76XBSfO1jjAY5kRVNXixwDEQ%3D%3D&X-Amz-Signature=20f8a8244a5ab4b61c072806f3117130e125f828fe47b2a9fa1b102d18affbdf) is enough:

Reply:
ahh the trace printed to stderr seems to be only for git-lfs, I think we would also need it in the file, we don't know what is causing connection issues, it may be lfs (also good to remove it fro the console).

Reply:
i'm not sure why the git-lfs trace isn't written to the file as well, but since it isn't I guess we need to go back to redirecting STDERR to a file

Reply:
yep sounds good